aws-lambda/05_UpdatingOurFunction.md

```diff
@@ -0,0 +1,94 @@
+# Step 4: Updating our function
+*(Full source code: [`step4`](step4) directory)*
+
+## /hello/`<whoever-you-are>`
+
+A function that returns `"Hello World!"` is not very useful.
+We'll make the returned Hello message [vary based on the URL path](https://quantummaid.de/docs/2_httpmaid/04_handlingrequests.html#request-route-and-path-parameters).
+
+<!---[CodeSnippet](step4HttpMaidConfig)-->
+```java
+private static QuantumMaid quantumMaidConfig() {
+  final QuantumMaid quantumMaid = QuantumMaid.quantumMaid()
+      .get("/hello/<whoever-you-are>", (request, response) -> response.setBody(
+          String.format("Hello %s!",
+              request.pathParameters().getPathParameter("whoever-you-are"))));
+  return quantumMaid;
+}
+```
+
+## Updating the function using SAM CLI
+
+To deploy the change, use the same command pair as in step 3.
+This time however, we don't need to answer any questions, because previous answers are saved to `samconfig.toml`
+
+```shell
+$ sam build
+$ sam deploy
+...
+Successfully created/updated stack - hello-app in us-east-1
+```
+
+## Viewing logs using SAM CLI
+
+Before we invoke the URL, let's tail the logs, which will give us some interesting details on our function execution(s).
+In a separate terminal, run:
+
+```shell
+$ sam logs --tail --region us-east-1 --stack-name hello-app --name HelloWorldFunction
+```
+
+## Getting runtime statistics
+
+Back in the terminal where you ran `sam deploy`, run:
+
+```shell
+$ time echo $(curl -s $apiUrl/hello/{me,you,everyone})
+Hello me!Hello you!Hello everyone!
+
+real	0m2.990s
+user	0m0.077s
+sys	0m0.005s
+```
+
+The logs display something similar to the following simplified output:
```

Comment:
Maybe it should be stated that it could take a bit of time until the logs arrive. This happened with me and I was confused.